File: JAVA/11_Brute_Force/BOJ_01_2798_블랙잭.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.BufferedWriter;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int card[] = new int[n];
        st = new StringTokenizer(br.readLine(), " ");
        for (int i = 0; i < n; i++) {
            card[i] = Integer.parseInt(st.nextToken());
        }
        int max = 0;
        for (int i = 0; i < n - 2; i++) {
            for (int j = i + 1; j < n - 1; j++) {
                for (int k = j + 1; k < n; k++) {
                    if (card[i] + card[j] + card[k] <= m && card[i] + card[j] + card[k] > max) {
                        max = card[i] + card[j] + card[k];
                    }
                }
            }
        }
        bw.write(max + "");
        bw.flush();
        bw.close();
    }
}

File: C/11_Brute_Force/BOJ_01_2798_블랙잭.c
#include <stdio.h>
main() {
	int n, m, card[100], max = 0, sum = 0;
	scanf("%d", &n);
	scanf("%d", &m);
	for (int i = 0; i < n; i++) {
		scanf("%d", &card[i]);
	}
	for (int i = 0; i < n; i++) {
		for (int j = i + 1; j < n; j++) {
			for (int k = j + 1; k < n; k++) {
				sum = card[i] + card[j] + card[k];
				if (sum <= m && sum > max) {
					max = sum;
				}
			}
		}
	}
	printf("%d", max);
}